task8/src/main/java/OneAd.java

```diff
@@ -0,0 +1,66 @@
+import AdCollection.Ad;
+import AdCollection.AdService;
+import com.google.gson.Gson;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+public class OneAd extends HttpServlet {
+
+    private AdService posts = AdService.getInstance();
+
+    @Override
+    protected void doDelete(HttpServletRequest request, HttpServletResponse response) throws ServletException {
+        String str = request.getQueryString();
+        String sID = str.substring(3, str.length());
+        int id = Integer.parseInt(sID);
+        response.getOutputStream().println("it Works id= " + id);
+        if (posts.removePost(id)) {
+            response.getOutputStream().println("deleted\n");
+        } else {
+            response.getOutputStream().println("Not found\n");
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String str = request.getQueryString();
```

Comment:
The "fool-proof" check is lacking. What if we do not have"id=" in the query string? Or the id is not a number?